src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -94,20 +94,54 @@ boolean isDirty() {
 
     public void openFile(@Nullable Long offset) throws IOException {
         SourceFile fileProps = new SourceFile();
-
         File folder = new File(basePath);
         if (!folder.exists() && !folder.mkdirs()) {
             if (!folder.exists()) {
                 throw new IOException(String.format("Failed to create new directory %s", folder.getPath()));
             }
             log.warn("Couldn't create the directory because it already exists (likely a race condition)");
         }
-
         String filePath = getFilePath.apply(offset);
         fileProps.path = filePath;
-        File file = new File(filePath);
+        // Sanitize the file name just be sure and make sure it has the R/W permissions only
+        String sanitizedFilePath = FilenameUtils.normalize(filePath);
+        if (sanitizedFilePath == null) {
+            /*
+             * This condition should not occur at all. The files are created in controlled manner with the names consisting DB name, table name. These do not
+             * permit names like "../../" or "./" etc. Still adding an additional check.
+             */
+            String errorMessage = String.format("Exception creating local file for write." +
+                    "File %s has non canonical path", filePath);
+            throw new RuntimeException(errorMessage);
+        }
+        File file = new File(sanitizedFilePath);
+        boolean createFile = file.createNewFile(); // if there is a runtime exception. It gets thrown from here
+        if (createFile) {
+            /*
+             * Setting restricted permissions on the file. If these permissions cannot be set, then warn, We cannot fail the ingestion. Added this in a

```

Comment:
Maybe explain why we cannot fail the ingestion - is it because we expect scenarios where the application does not have the permission to change set the restriction policies but this is not a reason to block the ingestion work?
```suggestion
             * Setting restricted permissions on the file. If these permissions cannot be set, then warn - We cannot fail the ingestion. Added this in a

```